Added overload for appSettings configuration to resolve #425

matthewrwilton opened the pull request:
Added a simple overload with the parameters @mivano suggested. I made the index format and template registration optional to remain in the spirit of bare minimum configuration (server address only) required.

If this gets merged in the "Breaking changes for version 2" section in the readme should probably be updated to reflect that basic appSettings configuration is possible.

mivano commented on src/Serilog.Sinks.Elasticsearch/LoggerConfigurationElasticSearchExtensions.cs:
```diff
@@ -56,5 +56,35 @@ public static class LoggerConfigurationElasticsearchExtensions
+        /// <returns>LoggerConfiguration object</returns>
+        public static LoggerConfiguration Elasticsearch(
+            this LoggerSinkConfiguration loggerSinkConfiguration,
+            string uri,
```
We can also pass in a Uri directly. This is supported by the AppSettings config (https://github.com/serilog/serilog/blob/dev/src/Serilog/Settings/KeyValuePairs/KeyValuePairSettings.cs#L150).

mivano commented on src/Serilog.Sinks.Elasticsearch/LoggerConfigurationElasticSearchExtensions.cs:
```diff
@@ -56,5 +56,35 @@ public static class LoggerConfigurationElasticsearchExtensions
+            string indexFormat = null,
+            string templateName = null)
+        {
+            var options = new ElasticsearchSinkOptions(new[] { new Uri(uri) });
```
Although if we use a string, we might be able to also allow passing in a collection or urls. So do a split on ; or , and pass those in as Uri array.

mivano commented:
Nice addition. See my inline comments. Not a big deal, but might be worthwhile to consider as changes afterwards are more difficult.

matthewrwilton commented:
I did wonder about adding support for multiple nodes and since you've brought it up it seems like it is worth adding. I'll add that in and resubmit.

matthewrwilton commented:
@mivano I added support for configuring multiple nodes using a comma separated list. I also realised that an exception could be thrown if the URI is invalid and I added handling to suppress this (which is how I think Serilog sinks are expected to behave).

mivano commented:
Thanks. I m just wondering if we should accept an incorrect Uri and silently use the default value when there is an exception. Normally the registration of a sink happens at startup and the other sinks throw an exception (see e.g. https://github.com/serilog/serilog/blob/dev/src/Serilog.FullNetFx/Sinks/RollingFile/RollingFileSink.cs#L64). So a misconfiguration can now lead to a localhost connection. Which might not be there and consequential logging will fail. The selflog is not very discoverable. 

@nblumhardt you have an opinion about this?

Besides this, a nice addition as most people only specify a (collection of) uri(s).

nblumhardt commented:
I'd favour consistency with the other sinks in this case - keeps things predictable.

matthewrwilton commented:
No worries. Updated to throw exceptions for invalid URIs.

mivano commented:
I merged this and a new package will be out soon. Documentation has been updated too. Thanks for the PR!

matthewrwilton commented:
Happy to help out :smile:
